src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -995,27 +1002,61 @@ private object GetValueFromSourceRow(int destRowIndex, out bool isSqlType, out b
                                 Debug.Assert(!(value is INullable) || !((INullable)value).IsNull, "IsDBNull returned false, but GetValue returned a null INullable");
                             }
 #endif
-                            return value;
+                            return WriteValueAsync(value, destRowIndex, isSqlType, isDataFeed, isNull);
                         }
                     }
                     else
                     {
-                        isDataFeed = false;
-
-                        IDataReader rowSourceAsIDataReader = (IDataReader)_rowSource;
+                        IDataReader r = (IDataReader)_rowSource;
```

Comment:
I don't say the previous name is the best choice but I'd rather a self-descriptive variable.

Reply:
reverted to old name to minimize flux